### ISSUE TITLE
Add per-tag whitespace control

### ISSUE THREAD

**Issue by ob-ivan** [twigjs/twig.js#108]
http://twig.sensiolabs.org/doc/templates.html#whitespace-control

Twig provides a special syntax for eliminating unnecessary whitespaces around tags:

```
<a long="attributes" and="all that stuff" unreadable="coding standard line length limit exceeded">
    {{- value -}}
    {%- if all is ok %}
        add some text
    {%- endif -%}
</a>
```

This feature is extremely useful when formatting template source code.

**Comment by nebulousGirl** [comment]
I am using Twig with Twig.js and PHP. I'd at least think the compiler should not choke when encountering the "-" which is currently the case. 

This way I can use a preprocessor for HTML versions using Twig.js, but I can use the built-in support in PHP with the same templates.

**Comment by TxHawks** [comment]
Any news on this? This is both an important feature and essential for protability

**Comment by ob-ivan** [comment]
@TxHawks Adding appropriate test cases seems to be a good place to start. Do you mind adding some?

Also, from the first glance at the work-around by @nebulousGirl the challenge with the straightforward implementation is that opening and closing tokens are not bound to be symmetrical: `{{- value }}` is as valid, as well as `{{ value -}}`.

**Comment by dave-irvine** [comment]
Fixed in #266
